engine/access/rpc/connection/connection.go

```diff
@@ -0,0 +1,142 @@
+package connection
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"time"
+
+	"github.com/onflow/flow/protobuf/go/flow/access"
+	"github.com/onflow/flow/protobuf/go/flow/execution"
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/module"
+)
+
+// ConnectionFactory is an interface for creating access and execution API clients.
+type ConnectionFactory interface {
+	GetAccessAPIClient(address string) (access.AccessAPIClient, io.Closer, error)
+	GetAccessAPIClientWithPort(address string, port uint) (access.AccessAPIClient, io.Closer, error)
+	InvalidateAccessAPIClient(address string)
+	GetExecutionAPIClient(address string) (execution.ExecutionAPIClient, io.Closer, error)
+	InvalidateExecutionAPIClient(address string)
+}
+
+// ProxyConnectionFactory wraps an existing ConnectionFactory and allows getting API clients for a target address.
+type ProxyConnectionFactory struct {
+	ConnectionFactory
+	targetAddress string
+}
+
+type noopCloser struct{}
+
+func (c *noopCloser) Close() error {
+	return nil
+}
```

Comment:
can you move these to `manager.go`. that's the only place that's currently using it.